R/deployApp.R

```diff
@@ -355,23 +355,29 @@ deployApp <- function(appDir = getwd(),
       target$appName,
       target$appTitle,
       "shiny",
-      accountDetails$accountId
+      accountDetails$accountId,
+      appMetadata$appMode
     )
     taskComplete(quiet, "Created application with id {.val {application$id}}")
   } else {
     application <- taskStart(quiet, "Looking up application with id {.val {target$appId}}...")
     application <- tryCatch(
-      client$getApplication(target$appId),
+      {
+        application <- client$getApplication(target$appId, target$version)
+        taskComplete(quiet, "Found application {.url {application$url}}")
+
+        if (application$type == "static") {
```

Comment:
This "static" notion is specific to Posit Cloud; does it belong here or with the call to `uploadCloudBundle` (where we already have cloud-specific logic)?

Related: Do we also need to call `createRevision` in the 404 path just below? `applicationDeleted` calls `createApplication` -- could you explain why this path explicitly creates a revision while the creation path does not?